tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -131,17 +135,38 @@ def setup_method(self):
             self.num_replica = int(resp[1])
         else:
             self.num_replica = 1
+        LOGGER.info("Create IAM user")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.s3_clean.update(users)
+        access_key = list(users.values())[0]["accesskey"]
+        secret_key = list(users.values())[0]["secretkey"]
+        self.s3_test_obj = S3TestLib(access_key=access_key, secret_key=secret_key,
+                                     endpoint_url=S3_CFG["s3_url"])
+        self.s3_ver = S3VersioningTestLib(access_key=access_key, secret_key=secret_key,
+                                          endpoint_url=S3_CFG["s3_url"], region=S3_CFG["region"])
+        LOGGER.info("Created IAM user")
+        if self.setup_type == "VM":
+            self.f_size = str(HA_CFG["5gb_mpu_data"]["file_size_512M"]) + "M"
+        else:
+            self.f_size = str(HA_CFG["5gb_mpu_data"]["file_size"]) + "M"
         LOGGER.info("COMPLETED: Setup operations. ")
 
     def teardown_method(self):
         """
         This function will be invoked after each test function in the module.
         """
+        LOGGER.info("STARTED: Teardown Operations.")
+        if self.version_etag:
+            for bucket in list(self.version_etag.keys()):
+                LOGGER.info("Deleting object versions for %s", bucket)
+                for v_etag in self.version_etag[bucket]:
+                    v_id = list(v_etag.keys())[0]
+                    self.s3_ver.delete_object_version(bucket=bucket, key=self.object_name,
+                                                      version_id=v_id)
```

Comment:
Deleting bucket itself doesn't take care of deleting all versions of object ? Do we have to delete each version one-by-one?